src/rust/cryptography-x509-validation/src/types.rs

```diff
@@ -69,6 +69,16 @@ impl<'a> DNSName<'a> {
             None => None,
         }
     }
+
+    /// Returns this DNS name's labels, in reversed order
+    /// (from top-level domain to most-specific subdomain).
+    fn rlabels(&self) -> impl Iterator<Item = &'_ str> {
+        self.as_str()
+            .split('.')
+            .collect::<Vec<_>>()
+            .into_iter()
+            .rev()
```

Comment:
Surely this can just be an rsplit?

Reply:
Yeah, thanks for catching that -- I'm surprised clippy didn't. Will fix.

Reply:
Fixed, and opened a Clippy request: https://github.com/rust-lang/rust-clippy/issues/11632